archetypes/mp-bare/src/main/resources/archetype-resources/src/main/java/Main.java.vm

```diff
@@ -0,0 +1,57 @@
+#set( $symbol_pound = '#' )
+#set( $symbol_dollar = '$' )
+#set( $symbol_escape = '\' )
+package ${package};
+
+import java.io.IOException;
+import java.util.logging.LogManager;
+
+import io.helidon.microprofile.server.Server;
+
+/**
+ * Main method simulating trigger of main method of the server.
+ */
+public final class Main {
+
+    /**
+     * Cannot be instantiated.
+     */
+    private Main() { }
+
+    /**
+     * Application main entry point.
+     * @param args command line arguments
+     * @throws IOException if there are problems reading logging properties
+     */
+    public static void main(final String[] args) throws IOException {
+#if( $loggingConfig.matches("y|yes|true") )
+        setupLogging();
+#end
+        Server server = startServer();
+#if( $restResource.matches("y|yes|true") )
+        System.out.println("http://localhost:" + server.port() + "${restResourcePath}");
+#end
+    }
+
+    /**
+     * Start the server.
+     * @return the created {@link Server} instance
+     */
+    static Server startServer() {
+        // Server will automatically pick up configuration from
+        // microprofile-config.properties
+        // and Application classes annotated as @ApplicationScoped
+        return Server.create().start();
+    }
+
+#if( $loggingConfig.matches("y|yes|true") )
+    /**
+     * Configure logging from logging.properties file.
+     */
+    private static void setupLogging() throws IOException {
+        // load logging configuration
+        LogManager.getLogManager().readConfiguration(
+                Main.class.getResourceAsStream("/logging.properties"));
```

Comment:
Use a try-with-resources to ensure this stream is not left open (`readConfiguration` doesn't close it; on Windows this will lock the jar).

Reply:
This is from the helidon-mp-quickstart, I'll make the change to both the archetype and the quickstart.